### ISSUE TITLE
Unable to manipulate cursor with input command

### ISSUE THREAD

**Issue by cotone-dev** [nushell/nushell#14274]
## Issue
While typing with the input command, I was hoping to be able to move the cursor left or right over the input string, but I am having trouble doing so.

```shell
> input
aaaaa| <-- The cursor is fixed at the end.
```

Is it possible to achieve cursor-moveable input, as in the `read -e` command?

## Environment
OS: MacOS 15.0.1
Terminal emulator: WezTerm 20240203-110809-5046fc22
Shell: Nushell 0.99.1

We would be grateful for your wisdom.

### How to reproduce

1. Run the input command on the terminal
2. Entering multiple characters
3. Verify that the cursor is fixed at the end and that the left and right keys do not move the cursor

### Expected behavior

After inputting multiple characters with the input command, the user should be able to move the cursor within the input string with the left and right keys, etc.

### Configuration

| key                | value                                          |
| ------------------ | ---------------------------------------------- |
| version            | 0.99.1                                         |
| major              | 0                                              |
| minor              | 99                                             |
| patch              | 1                                              |
| branch             |                                                |
| commit_hash        |                                                |
| build_os           | macos-x86_64                                   |
| build_target       | x86_64-apple-darwin                            |
| rust_version       | rustc 1.81.0 (eeb90cda1 2024-09-04) (Homebrew) |
| cargo_version      | cargo 1.81.0                                   |
| build_time         | 2024-10-20 02:45:05 +00:00                     |
| build_rust_channel | release                                        |
| allocator          | mimalloc                                       |
| features           | default, sqlite, trash                         |
| installed_plugins  |                                                |

**Comment by petricavalry** [comment]
Thanks for reporting this.

https://github.com/nushell/nushell/blob/c7e128eed1358c86bb2878b46a519ffab30144d5/crates/nu-command/src/platform/input/input_.rs#L96-L125

It doesn't handle [KeyCode::Right](https://docs.rs/crossterm/latest/crossterm/event/enum.KeyCode.html#variant.Right) and [KeyCode::Left](https://docs.rs/crossterm/latest/crossterm/event/enum.KeyCode.html#variant.Left).

**Comment by cotone-dev** [comment]
Thank you for informing us about the implementation of the input command.
I see that is the current specification.

I would like to ask you about the following

1. Are you planning to implement the left and right keys in the future? If not, we would be glad to know why not.
2. Is there a nushell recommended implementation of something like `read -e` that allows cursor movement?

**Comment by petricavalry** [comment]
Sorry. I'm still learning Rust.

I'm willing to help. I'm unsure how to handle variable-length characters.

Try [Zenity](https://gitlab.gnome.org/GNOME/zenity) and [KDialog](https://github.com/KDE/kdialog).

**Comment by cotone-dev** [comment]
Thank you for your prompt reply.
Excuse me, is petricavalry a member of nushell management?

**Comment by petricavalry** [comment]
You can see label "Contributor", "Member" and "Author" in top right corner of comment. I am none of those (in this issue).

Sorry for misunderstanding.

**Comment by cotone-dev** [comment]
Thank you very much.
My apologies for the mix-up as well.

We will wait for a response from the "Member" or "Contributor".
Thank you very much for the information you provided.

**Comment by cotone-dev** [comment]
@fdncred 
Can you please respond to this question when you have time?

**Comment by fdncred** [comment]
> Is it possible to achieve cursor-moveable input, as in the read -e command?

No, it's not possible at this time. However, we'd probably accept a PR to allow it.

**Comment by cotone-dev** [comment]
Thank you for your prompt response.

I understand that this is currently not possible.

However, I have no development experience with Rust.
Could you please schedule an implementation with your development team?

**Comment by fdncred** [comment]
If someone wants to pick this up, please do so.

We're all volunteers so there's little to no scheduling of people to work on anything.

**Comment by cotone-dev** [comment]
Thank you.
I see!

I like nushell a lot and would like to contribute if I can.
I would like to learn Rust in the future and create a PR.

In the meantime, modifications by others are of course welcome.

Can I leave the issue as is?
